### ISSUE TITLE
[BUG] PixelDrain API key is ignored?

### ISSUE THREAD

**Issue by baccccccc** [Jules-WinnfieldX/CyberDropDownloader#830]
I have the API key specified, which is confirmed by downloader.log: `"PixelDrain Provided": true,`.

My PD transfer limit is nowhere near depletion. According to the PD account page, my transfer limit is 4 TB, and I've only used 24 GB in the past month.

Nevertheless, I'm hitting a ton of 403's when downloading from PixelDrain. If I retry the same link in browser, it works.

Also, PD account page shows almost no bandwidth usage for today, even though I was constantly trying to download stuff.

![image](https://github.com/Jules-WinnfieldX/CyberDropDownloader/assets/107458411/b2509b8b-f67f-4d3d-b441-4c57389e839d)

This altogehter makes me think that the PixelDrain API key is somehow mishandled. E.g., either it's stored or transmitted in the wrong format, or maybe ignored altogehter.

Thoughts?

**Comment by baccccccc** [comment]
after watching it more closely for a day, it looks like I am *sometimes* using the “premium bandwidth”.

dunno, perhaps CDL is not guilty, it's just PD randomly decides whether to honor the API key vs. bouncing me back with a 403 ¯\_(ツ)_/¯

![image](https://github.com/Jules-WinnfieldX/CyberDropDownloader/assets/107458411/f8491ba3-149c-4b0e-bd47-0a4b5edf4b44)

**Comment by Jules-WinnfieldX** [comment]
Weird indeed. I'll still look into it, but I'm trying to squash more major things at the moment. I'll also need to buy a new premium key to test, so... holding off for the moment.

**Comment by VirginCrusader** [comment]
I did have this problem on Windows 10, however i resolved it by uninstalling python and switching to Windows Terminal. I also upgraded my Powershell to 7. i also deleted any leftovers of Cyberdrop-dl before installing it again. It now works correctly and it uses my Pixeldrain api. What i noticed when the problem occurred -> Cyberdrop-dl for whatever reason had reverted back to 4.x instead of the upgraded version 5.x, without me taking any steps to do so.
